python/repositories.bzl

```diff
@@ -193,15 +205,17 @@ def _python_repository_impl(rctx):
         if "windows" not in platform:
             lib_dir = "lib" if "windows" not in platform else "Lib"
 
+            python = _get_host_python_interpreter(rctx)
```

Comment:
I think we should use the `python_bin` var that is defined below instead of using the `host_python`. And we should do the `read-only` fix only if the `host` platform the same as the `platform` of the toolchain.

Could we recursively change this for `windows` as well? Right now we are calling this only if it is not Windows, but do the scripts work on Windows?

Reply:
[os.chmod](https://docs.python.org/3/library/os.html#os.chmod) says:

> Although Windows supports [chmod()](https://docs.python.org/3/library/os.html#os.chmod), you can only set the file’s read-only flag with it (via the `stat.S_IWRITE` and `stat.S_IREAD` constants or a corresponding integer value). All other bits are ignored.

So, yes, it would remove write access on Windows.

Reply:
The `chmod.py` _does_ set the folders and files to read-only on Windows, but does not prevent new files being created in the directories. So the same behaviour as Unix is not present. We would have to look into Windows permissions to restrict that.

The `getuid` could return the Window Security ID for the user but that doesn't really help much. I've removed the `touch` check by checking `platform`.

We now run the scripts on Windows, but they don't really do much and actually fail to prevent `.pyc` files being written. Which is no different to before this patch.

Reply:
One can prevent writing files into directories on Windows with security permissions. That could be done by calling out to `icacls.exe` (in leiu of the `win32security` dependency). That will likely take significant extra testing as I'm not a Windows expert.

Reply:
@aignas would you like me to continue figuring out truely read-only directories on Windows or does this PR move the needle enough to be considered for merge? I'd be happy to raise follow up issues/PRs, if necessary.